plugins/hls-stan-plugin/src/Ide/Plugin/Stan.hs

```diff
@@ -164,24 +164,25 @@ rules recorder plId = do
               logWith recorder Debug (LogDebugStanEnvVars env)
               seTomlFiles <- liftIO $ usedTomlFiles useDefConfig (stanArgsConfigFile stanArgs)
 
+              -- Note that Stan works in terms of relative paths, but the HIE come in as absolute. Without
+              -- making its path relative, the file name(s) won't line up with the associated Map keys.
+              relativeHsFilePath <- liftIO $ makeRelativeToCurrentDirectory $ fromNormalizedFilePath file
+              let hieRelative = hie{hie_hs_file=relativeHsFilePath}
+
               (cabalExtensionsMap, checksMap, confIgnored) <- case configTrial of
                   FiascoL es -> do
                       logWith recorder Development.IDE.Warning (LogWarnConf es)
                       pure (Map.empty,
                             HM.fromList [(LSP.fromNormalizedFilePath file, inspectionsIds)],
                             [])
                   ResultL warnings stanConfig -> do
-                      let currentHSAbs = fromNormalizedFilePath file -- hie_hs_file hie
-                      currentHSRel <- liftIO $ makeRelativeToCurrentDirectory currentHSAbs
-                      cabalExtensionsMap <- liftIO $ createCabalExtensionsMap isLoud (stanArgsCabalFilePath stanArgs) [hie]
-
-                      -- Files (keys) in checksMap need to have an absolute path for the analysis, but applyConfig needs to receive relative
```

Comment:
> Files (keys) in checksMap need to have an absolute path for the analysis

This didn't _seem_ to be the case, when I looked?

I think this logic was added in https://github.com/haskell/haskell-language-server/pull/3914 ... perhaps @0rphee could confirm?

Reply:
Yes this was added in that PR. As I remember it, If I used relative paths, it didn't work, because of how an internal function works in `runAnalysis`, that checks if an absolute path to a file, begins with another filepath, (`currentHSAbs` I think?).